File: test/ruby/spec.rb
require 'oci8'

RSpec.describe 'library access' do
  it 'can access without password as the current user' do
    rows_returned = false

    OCI8.new('/').exec('SELECT 1 FROM DUAL') do |row|
      rows_returned = true
      expect(row).to eq([1])
    end

    expect(rows_returned).to be true
  end
end

File: test/node/spec.js
'use strict';

var expect = require('chai').expect;
var oracledb = require('oracledb');

describe('library access', function () {
  it('can access without password as the current user', function () {
    oracledb.getConnection({connectString: '/'}, function (err, connection) {
      expect(err).to.not.exist;

      connection.execute('SELECT 1 FROM DUAL', function (err, result) {
        expect(err).to.not.exist;
        expect(result.rows).to.equal([[1]]);
      });
    });
  });
});
